### DISCUSSION TITLE
speed limit gaps for highway=motorway in the USA and Canada

### DISCUSSION THREAD

**Discussion by karussell** [westnordost/osm-legal-default-speeds#17] [General]
There are situations where the wiki is incomplete for the USA and Canada, at least for highway=motorway. Some of the gaps are obvious like when there is no speed limit defined for "US interstate highway" like for California but especially for urban cases [where the library currently returns 25mph](https://westnordost.github.io/osm-legal-default-speeds/#tags=highway%3Dmotorway%0Arural%3Dno&cc=US-CA). Or for `highway=motorway` without `ref=I.*` it might be that there is no default speed limit defined. 

Reading through the description e.g. in wikipedia ([here](https://en.wikipedia.org/wiki/Speed_limits_in_the_United_States_by_jurisdiction) and [here](https://en.wikipedia.org/wiki/Speed_limits_in_the_United_States#)) does not make it clear to me. Especially the urban speed limit for freeways/interstates. Another example is [Texas](https://en.wikipedia.org/wiki/Speed_limits_in_the_United_States_by_jurisdiction#Texas) for which [the library calculates 30mph](https://westnordost.github.io/osm-legal-default-speeds/#tags=highway%3Dmotorway%0Arural%3Dno&cc=US-TX) at the moment. 

A similar problem exists [for freeways in Canada](https://en.wikipedia.org/wiki/Speed_limits_in_Canada).

For the US states that have no value for "motorway" in the wiki, I found the following values ("state, urban max, rural max")

* Arizona: 55-65?, 75
    example for "urban interstate": https://www.openstreetmap.org/way/37442414
* Arkansas: 65, 75
    Along rural freeways the limit is 75 mph (121 km/h) while suburban freeways are posted at 65 mph (105 km/h).
* California: 55?, 70
* Delaware: 50-65 ?
* District of Columbia: 40-55?, no rural freeways
* Georgia: 55-70, 70
* Hawaii: 45–50, 55–60
* Indiana: 55?, 70
* Kansas: 40, 75
* Massachusetts: urban?, 65
* Maryland: urban?, 70
* Maine: urban?, 70
* Mississippi: urban?, 70
* North Carolina: urban?, 65-70
* North Dakota: urban?, 75
* New Jersey: urban?, 65
* New Mexico: urban?, 75
* (Nevada: urban?, 70–80)
* New York: urban?, 65
* Oklahoma: 45-60, 70-80
* Pennsylvania: urban?, 55–70
* Rhode Island: urban?, 65
* South Carolina: 45, 65-70
* (Tennessee: urban?, 65-70)
* Texas: urban?, 75-85
* Utah: 45, 75-80
* Vermont: urban?, 65
* West Virginia: urban? 70

Thinking about this more ... even if we can fix these values in the wiki maybe in some situations it is never possible to return a specific default speed. Maybe the library should be able to return an  "uncertain" value for specific road types too? E.g. then it would be easy to map "urban motorways" in Texas to this result and know when to ignore the values from the library.

**Update:** [Now a workaround in GraphHopper exists](https://github.com/graphhopper/graphhopper/commit/6a4e1a81d34a9757088b4e3a06a38b9207173d67) until these values are added in the wiki.

**Comment by westnordost** [comment]
Well, what's in the wiki is the max speed as defined in the law. It does not list the "usually to be expected" speed limits.

> Another example is [Texas](https://en.wikipedia.org/wiki/Speed_limits_in_the_United_States_by_jurisdiction#Texas) for which [the library calculates 30mph](https://westnordost.github.io/osm-legal-default-speeds/#tags=highway%3Dmotorway%0Arural%3Dno&cc=US-TX) at the moment.

I think a good mitigation would be to never consider `highway=motorway` as "urban". I am assuming that the `rural=no` actually comes from your data source that assigns `rural=yes/no` to every road by analyzing the density of roads. So, that algorithm should rather never assign `urban=yes` to any `highway=motorway or expressway=yes or motorroad=yes` (i.e. any access-controlled road)

**Comment by karussell** [comment]
> I think a good mitigation would be to never consider highway=motorway as "urban"

This would make a good workaround. To my understanding the problem is still that there are cases where there are legal restrictions for urban interstates. Or are these values only observations and no laws?

But even the OSM wiki already has a few candidates like Indiana, Missouri, Montana and Minnesota where it makes a difference between a rural and an urban interstate.

**Reply by westnordost** [reply]
Yes, there are some legal restrictions for urban motorways. Look for `urban motorway` in the table. Note that interstates are not necessarily motorways, they are frequently trunk roads, too, i.e. have level intersections.

On how to handle `urban motorway`s - hm, I would expect that the speed limit in those sections of motorway are properly signed (i.e. not just a "Welcome to $City" but a proper maxspeed sign). Especially in the United States, where city limits are usually irrelevant for determining whether a particular road segment is within an "urban area" or not - in the US, "urban area" is usually defined by fuzzy properties such as how many houses per 100m are next to the road.

If however the motorway is tagged as urban (maxspeed/traffic zone = urban) in OSM, I would trust this value. The "do not tag motorways with urban=yes/no" workaround should only apply to the algorithm you are using, not to how to handle tags already present in OSM.

**Reply by karussell** [reply]
> Note that interstates are not necessarily motorways, they are frequently trunk roads, too, i.e. have level intersections.

Can you explain what you mean here? As "US interstate highway" is defined as `ref~I.* and {motorway}` in the wiki.

**Reply by westnordost** [reply]
Oh, hmm... and the footnote links to https://wiki.openstreetmap.org/wiki/Interstate_Highways .
I guess they are all motorways, then, I might misremember. I asked in the slack US chat.

**Reply by westnordost** [reply]
I asked in the chat, indeed there are sections of Interstates that are not motorways, see https://en.wikipedia.org/wiki/List_of_gaps_in_Interstate_Highways#At-grade_intersections

However, these seem to be not tagged as `highway=motorway` then, but `highway=trunk`. See this overpass query:
https://overpass-turbo.eu/s/1FHI
